cfgov/v1/models/base.py

```diff
@@ -255,6 +267,37 @@ def get_context(self, request, *args, **kwargs):
         if self.schema_json:
             context['schema_json'] = self.schema_json
 
+        context['meta_description'] = ''
+        if self.search_description:
+            context['meta_description'] = self.search_description
+        if hasattr(self, 'header') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.header,
+                'hero',
+                'body')
+        if (
+            hasattr(self, 'preview_description')
+            and not context['meta_description']
+        ):
+            context['meta_description'] = self.preview_description
+        if hasattr(self, 'header') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.header,
+                'text_introduction',
+                'intro')
+        if hasattr(self, 'content') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.content,
+                'text_introduction',
+                'intro')
+        if hasattr(self, 'header') and not context['meta_description']:
+            context['meta_description'] = self.get_streamfield_content(
+                self.header,
+                'item_introduction',
+                'paragraph')
+        if not context['meta_description']:
```

Comment:
Already set at 270

Reply:
But I see your test for blank meta_description fails without it.

Reply:
Ah -- the tests for existence of `header` or `content` then dive into each field's streamfield, which can return None, thus overriding line 270.

Rather than repeating tests for existence, It might be nicer to have a separate function that checks the various fields for possible meta_descriptions and returns either a winner or ''.

Reply:
@higs4281 I'm not sure what this would look like that would be any more elegant than this brute-force approach. Can you sketch it out in pseudocode?

Reply:
The `context['meta_description'] = ''` on L270 _is_ unnecessary, though, given this check at the end.